src/backend/concurrency/epoch_manager.h

```diff
@@ -206,29 +209,28 @@ class EpochManager {
       }
 
       // save max cid
-      auto max = epoch_queue_[idx].max_cid_;
-      AtomicMax(&max_cid_gc_, max);
+      auto max_ = std::max(epoch_queue_[idx].max_cid_, epoch_queue_[idx].max_readonly_cid_);
```

Comment:
I think GC tail should not consider the max_cid_ of the epoch. It only need to look at the max_read_only_cid_ and get the max.